src/core/dev/doca/ib_ctx_handler_doca.cpp

```diff
@@ -0,0 +1,275 @@
+/*
+ * SPDX-FileCopyrightText: NVIDIA CORPORATION & AFFILIATES
+ * Copyright (c) 2001-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "util/vtypes.h"
+#ifndef DEFINED_DPCP_PATH_RX_AND_TX
+#include "vlogger/vlogger.h"
+#include <util/sys_vars.h>
+#include "dev/ib_ctx_handler.h"
+#include "dev/time_converter_rtc.h"
+#include "sock/sock-app.h"
+
+#define MODULE_NAME "ibch"
+DOCA_LOG_REGISTER(ibch);
+
+#define ibch_logpanic   __log_panic
+#define ibch_logerr     __log_err
+#define ibch_logwarn    __log_warn
+#define ibch_loginfo    __log_info
+#define ibch_logdbg     __log_info_dbg
+#define ibch_logfunc    __log_info_func
+#define ibch_logfuncall __log_info_funcall
+
+ib_ctx_handler_doca::ib_ctx_handler_doca(doca_devinfo *devinfo, std::string &ibname,
+                                         const char *ifname)
+    : m_ifname(ifname)
+    , m_ibname(ibname)
```

Comment:
we save a bit of memory by having m_ibname by ref - but if ib_ctx_handler dtor logic is changed just a bit (e.g. - unique_ptr instead of new and default dtor) we're toast.

Reply:
It will be changes to receive the parent reference in the leftovers effort

Reply:
doca_devinfo &devinfo
Move these ones to be non pointers and use 3 constructors

Reply:
Done